ganttproject/src/net/sourceforge/ganttproject/GPTreeTableBase.java

```diff
@@ -528,10 +544,139 @@ protected void initTreeTable() {
     doInit();
   }
 
+  private class SortTableHeaderRenderer implements TableCellRenderer {
+    private final Icon myAscIcon;
+    private final Icon myDescIcon;
+    private final TableCellRenderer myDefaultRenderer;
+    SortTableHeaderRenderer(JTable t) {
+      myDefaultRenderer = t.getTableHeader().getDefaultRenderer();
+      myAscIcon = UIManager.getIcon("Table.ascendingSortIcon");
+      myDescIcon = UIManager.getIcon("Table.descendingSortIcon");
+    }
+
+    @Override
+    public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int col) {
+      Component c = myDefaultRenderer.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, col);
+      ColumnImpl column = myTableHeaderFacade.findColumnByViewIndex(col);
+      if (column.getSort() == SortOrder.ASCENDING) {
+        if (c instanceof JLabel) {
+            ((JLabel) c).setIcon(myAscIcon);
+        }
+      }
+      if (column.getSort() == SortOrder.DESCENDING) {
+        if (c instanceof JLabel) {
+          ((JLabel) c).setIcon(myDescIcon);
+        }
+      }
+      return c;
+    }
+  }
+
+  private <T>Comparator<T> reverseComparator(final Comparator<T> comparator) {
+    return new Comparator<T>() {
+      @Override
+      public int compare(T t1, T t2) {
+        return -comparator.compare(t1, t2);
+      }
+    };
+  }
+
+  private final TaskListener myRemoveOrderListener = new TaskListener() {
+
+    private void removeOrder() {
+      for (ColumnImpl c : myTableHeaderFacade.getColumns()) {
+        c.setSort(SortOrder.UNSORTED);
+      }
+    }
+
+    @Override
+    public void taskScheduleChanged(TaskScheduleEvent e) {
+      removeOrder();
+    }
+
+    @Override
+    public void dependencyAdded(TaskDependencyEvent e) {
+      removeOrder();
+    }
+
+    @Override
+    public void dependencyRemoved(TaskDependencyEvent e) {
+      removeOrder();
+    }
+
+    @Override
+    public void dependencyChanged(TaskDependencyEvent e) {
+      removeOrder();
+    }
+
+    @Override
+    public void taskAdded(TaskHierarchyEvent e) {
+      removeOrder();
+    }
+
+    @Override
+    public void taskRemoved(TaskHierarchyEvent e) {
+      removeOrder();
+    }
+
+    @Override
+    public void taskMoved(TaskHierarchyEvent e) {
+      removeOrder();
+    }
+
+    @Override
+    public void taskPropertiesChanged(TaskPropertyEvent e) {
+      removeOrder();
+    }
+
+    @Override
+    public void taskProgressChanged(TaskPropertyEvent e) {
+      removeOrder();
+    }
+
+    @Override
+    public void taskModelReset() {
+      removeOrder();
+    }
+  };
+
   protected void doInit() {
     setRootVisible(false);
     myCustomPropertyManager.addListener(this);
 
+    tableHeader.setDefaultRenderer(new SortTableHeaderRenderer(getTable()));
```

Comment:
Normally you want to access protected members from the superclass by calling accessor function, like you do one line below